packages/custom_lint/lib/src/v2/custom_lint_analyzer_plugin.dart

```diff
@@ -25,46 +27,64 @@ class CustomLintServer {
     required this.watchMode,
     required this.includeBuiltInLints,
     required this.delegate,
+    required this.workingDirectory,
   });
 
   /// Start the server while also capturing prints and errors.
-  static R? run<R>(
-    R Function(CustomLintServer server) cb, {
+  static Future<CustomLintServer> start({
     required SendPort sendPort,
     required bool watchMode,
     required bool includeBuiltInLints,
     required CustomLintDelegate delegate,
+    required Directory workingDirectory,
   }) {
     late CustomLintServer server;
-    final result = runZonedGuarded(
+
+    return runZoned(
+      () => server,
       () {
         server = CustomLintServer._(
           watchMode: watchMode,
           includeBuiltInLints: includeBuiltInLints,
           delegate: delegate,
+          workingDirectory: workingDirectory,
         );
         server._start(sendPort);
 
-        return cb(server);
+        return server;
+      },
+    );
+  }
+
+  /// Run the given [body] in a zone that captures errors and prints and
+  /// sends them to the server for handling.
```

Comment:
```suggestion
  /// sends them to the server for handling.
  ///
  /// Do not close the server within [runZoned], as this could cause a race condition
  /// on errors/prints handling, where an error/print happens after the server is closed,
  /// causing the event to be silenced.
```